### PR TITLE
Suite refactor

### PR DESCRIPTION
**Pull request by zmc**
This PR:
* Moves a lot of ``suite``'s functionality into a new ``Run`` class
* Makes use of ``GitbuilderProject``
* Makes ``suite`` a subpackage

It doesn't intend to change behavior otherwise.

### CODE REVIEW

**Comment by zmc** [comment]
http://pulpito.ceph.com/zack-2016-06-03_14:08:14-smoke-jewel---basic-vps/

File: teuthology/suite/run.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+                verbose=self.args.verbose,
+                log_prefix="Results email: ",
+            )
+            results_url = get_results_url(self.base_config.name)
+            if results_url:
+                log.info("Test results viewable at %s", results_url)
+
+    def schedule_suite(self):
+        """
+        Schedule the suite-run. Returns the number of jobs scheduled.
```
Comment (dmick) [inline teuthology/suite/run.py]: no hyphen necessary here

Comment (zmc) [inline teuthology/suite/run.py]: Really? That's how everyone seems to use the term. You'd prefer "suite run" ?

File: teuthology/suite/test/test_init.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    suite.Run.WAIT_PAUSE = 1
+
+    in_progress = deepcopy(results)
+    assert 0 == suite.wait('name', 1, 'http://UPLOAD_URL')
+    assert m_get_jobs.called_with('name', fields=['job_id', 'status'])
+    assert 0 == len(in_progress)
+    assert 'fail http://UPLOAD_URL/name/2' in caplog.text()
+
+    in_progress = deepcopy(results)
+    in_progress = deepcopy(results)
```
Comment (dmick) [inline teuthology/suite/test/test_init.py]: Is this subtle, or an error?

File: teuthology/suite/test/test_init.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    results.append([{'status': 'queued', 'job_id': '2'}])
+
+    def get_jobs(name, **kwargs):
+        return results.pop(0)
+    m_get_jobs.side_effect = get_jobs
+    suite.Run.WAIT_PAUSE = 1
+    suite.Run.WAIT_MAX_JOB_TIME = 1
+    with pytest.raises(suite.WaitException) as error:
+        suite.wait('name', 1, None)
+        assert 'abc' in str(error)
```
Comment (dmick) [inline teuthology/suite/test/test_init.py]: I don't get this line at all?...

File: teuthology/suite/util.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+    return gp.sha1
+
+
+def get_distro_defaults(distro, machine_type):
+    """
+    Given a distro (e.g. 'ubuntu') and machine type, return:
+        (arch, release, pkg_type)
+
+    This is used to default to:
+        ('x86_64', 'trusty', 'deb') when passed 'ubuntu' and 'plana'
```
Comment (dmick) [inline teuthology/suite/util.py]: indenting

File: teuthology/suite/test/test_init.py (hunk 2 above)
Comment (zmc) [inline teuthology/suite/test/test_init.py]: That certainly looks like an error!

File: teuthology/suite/run.py (hunk 1 above)
Comment (dmick) [inline teuthology/suite/run.py]: yeah.  it's a run, and the type of run is suite.  clearly a nit.

File: teuthology/suite/test/test_init.py (hunk 3 above)
Comment (zmc) [inline teuthology/suite/test/test_init.py]: Me either; I hadn't noticed that before. I wonder what it does...

File: teuthology/suite/util.py (hunk 4 above)
Comment (zmc) [inline teuthology/suite/util.py]: Good catch

**Pull request merged**